recipes/lzip/all/conanfile.py

```diff
@@ -0,0 +1,97 @@
+from conans import AutoToolsBuildEnvironment, CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+import textwrap
+
+
+class LzipConan(ConanFile):
+    name = "lzip"
+    description = "Lzip is a lossless data compressor with a user interface similar to the one of gzip or bzip2"
+    topics = ("conan", "lzip", "compressor", "lzma")
+    license = "GPL-v2-or-later"
+    homepage = "https://www.nongnu.org/lzip/"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = "patches/**"
+    settings = "os", "arch", "compiler", "build_type"
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Visual Studio is not supported")
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def _detect_compilers(self):
+        tools.rmdir("detectdir")
+        tools.mkdir("detectdir")
+        with tools.chdir("detectdir"):
+            tools.save("CMakeLists.txt", textwrap.dedent("""\
+                cmake_minimum_required(VERSION 2.8)
+                project(test C CXX)
+                message(STATUS "CC=${CMAKE_C_COMPILER}")
+                message(STATUS "CXX=${CMAKE_CXX_COMPILER}")
+                file(WRITE cc.txt "${CMAKE_C_COMPILER}")
+                file(WRITE cxx.txt "${CMAKE_CXX_COMPILER}")
+                """))
+            CMake(self).configure(source_folder="detectdir", build_folder="detectdir")
+            cc = tools.load("cc.txt").strip()
+            cxx = tools.load("cxx.txt").strip()
+        return cc, cxx
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH") and \
+                tools.os_info.detect_windows_subsystem() != "msys2":
+            self.build_requires("msys2/20190524")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+
+    @contextmanager
+    def _build_context(self):
+        env = {}
+        cc, cxx = self._detect_compilers()
+        if not tools.get_env("CC"):
+            env["CC"] = cc
+        if not tools.get_env("CXX"):
+            env["CXX"] = cxx
+        with tools.environment_append(env):
+            yield
```

Comment:
Do you think that the `AutoToolsBuildEnvironment` should populate always this environment variables?

Reply:
...my question is because I see this `_detect_compiler` like something for Conan, not for a recipe

Reply:
I had to do this hack to get/set `CC` and `CXX` because minizip is not really a autotools project.
Its configure script is manually written.
If it would have been autotools, this `_detect_compiler` wouldn't be necessary.

In general, doing this in conan can fix some potential bugs with profiles.
If a profile  has `settings.compiler=clang`, but doesn't set `env.CC`, then autodetect scripts might use cc (which probably is cc).
But I wouldn't do it in general, because *the linux way* is to use `/usr/bin/cc` and `/usr/bin/c++` by default.

Reply:
That said, it might be useful to add checks/hooks to the build helpers.
e.g. a hook that checks whether the detected compiler is really a compiler conforming to the profile.
But that's an entire other thing.

Reply:
Yes, that sounds like something for a [_binary linter_ ](https://github.com/conan-io/hooks/blob/master/hooks/binary_linter.py).